### PR TITLE
Migrate from circle CI to Github CI ++ Run tests on K8s

### LINKED ISSUE #125: Migrate Spring Boot example to use GitHub CI
Only in branch sb-2.4.x

### LINKED ISSUE #128: Run Spring Boot example on Kubernetes
Only for branch `sb-2.4.x`

### PR DESCRIPTION
**Pull request by Sgitario**
- Remove CircleCI scripts
- Remove project from CircleCI (otherwise Circle keeps triggering the builds...)
- Configure GIthub Action to run the tests on:
  - OpenShift: using s2i and Dekorate approaches.
  - Kubernetes: use Kind with internal registry

Fix https://github.com/snowdrop/rest-http-example/issues/125
Fix https://github.com/snowdrop/rest-http-example/issues/128

### CODE REVIEW

**Comment by Sgitario** [comment]
No needed anylonger.

**Comment by cmoulliard** [comment]
As the github openshift relies on a old release of ocp (= version 3), I'm wondering what could be the alternatives to test upstream using okd4:

1. Customize a "github action" to create the cluster using and okd installer (https://www.okd.io/installation/)
2. Use a "okd4 github action" if it exists 
3. Deploy on RHOS - https://docs.okd.io/latest/installing/installing_openstack/installing-openstack-installer-custom.html#installing-openstack-installer-custom but then How can a github action access such a service running behind a VPN
4. ???

@Sgitario

**Comment by Sgitario** [comment]
Did you have a chance to review this PR? @cmoulliard @jacobdotcosta

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  linux-build-local:
+    name: PR - Linux - JVM ${{ matrix.java }} - Local
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 8 ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Cache .m2 registry
```
Comment (metacosm) [inline .github/workflows/pr.yml]: Can't you do that with the setup-java action now?

Comment (Sgitario) [reply .github/workflows/pr.yml]: Didn't know we can cache the Maven dependencies using the `setup-java` action. Updated.

**Comment by cmoulliard** [comment]
I'm not a github action expert but If the change  works, then go ahead to merge it @Sgitario

**Pull request merged**